src/api/createDeviceReferenceDataASActions/tests/test_index.py

```diff
@@ -0,0 +1,160 @@
+        assert device_reference_data.questionnaire_responses == {}
+
+        # Retrieve the created resource
+        repo = DeviceReferenceDataRepository(
+            table_name=TABLE_NAME, dynamodb_client=index.cache["DYNAMODB_CLIENT"]
+        )
+
+        created_device_reference_data = repo.read(
+            product_team_id=device_reference_data.product_team_id,
+            product_id=device_reference_data.product_id,
+            device_reference_data_id=device_reference_data.id,
+        )
+        assert created_device_reference_data == device_reference_data
+
+
+def test_index_with_questionnaire() -> None:
+    questionnaire_data = [
+        {"Interaction ID": "foo"},
+        {"Interaction ID": "bar"},
+    ]
+
+    with mock_product() as (index, product):
+        # Execute the lambda
+        response = index.handler(
+            event={
+                "headers": {"version": VERSION},
+                "body": json.dumps(
+                    {
+                        "questionnaire_responses": {
+                            "spine_as_additional_interactions": questionnaire_data
+                        }
+                    }
+                ),
+                "pathParameters": {
+                    "product_team_id": str(product.product_team_id),
+                    "product_id": str(product.id),
+                },
+            }
+        )
+
+        # Validate that the response indicates that a resource was created
+        assert response["statusCode"] == 201
+
+        _device_reference_data = json_loads(response["body"])
+        device_reference_data = DeviceReferenceData(**_device_reference_data)
+        assert device_reference_data.product_id == product.id
+        assert device_reference_data.product_team_id == product.product_team_id
+        assert device_reference_data.name == "AAA-100001 - AS Additional Interactions"
+        assert device_reference_data.ods_code == ODS_CODE
+        assert device_reference_data.created_on.date() == datetime.today().date()
+        assert device_reference_data.updated_on.date() == datetime.today().date()
+        assert device_reference_data.deleted_on is None
+
+        questionnaire_responses = device_reference_data.questionnaire_responses[
+            "spine_as_additional_interactions/1"
+        ]
+        _questionnaire_data = [qr.data for qr in questionnaire_responses]
+        assert _questionnaire_data == questionnaire_data
+
+        # Retrieve the created resource
+        repo = DeviceReferenceDataRepository(
+            table_name=TABLE_NAME, dynamodb_client=index.cache["DYNAMODB_CLIENT"]
+        )
+
+        created_device_reference_data = repo.read(
+            product_team_id=device_reference_data.product_team_id,
+            product_id=device_reference_data.product_id,
+            device_reference_data_id=device_reference_data.id,
+        )
+        assert created_device_reference_data == device_reference_data
```

Comment:
Maybe add a test for invalid questionnaire responses, e.g. answered the wrong questionnaire, or supplied invalid answers to the spine_as_additional_interactions questionnaire?

Reply:
Ah can see all these scenarios are covered in the feature tests!